android/src/main/kotlin/es/antonborri/home_widget/HomeWidgetPlugin.kt

```diff
@@ -205,6 +217,10 @@ class HomeWidgetPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
         private const val WIDGET_INFO_KEY_ANDROID_CLASS_NAME = "androidClassName"
         private const val WIDGET_INFO_KEY_LABEL = "label"
 
+        // For example, both Double and Long are saved in SharedPreferences with putLong,
+        // and this Key is used to distinguish between them when retrieving.
+        private const val ACTUAL_WIDGET_DATA_TYPE = "actualWidgetDataType"
```

Comment:
This would never work when Users wanted to store both double and long as this would always override

Reply:
Oh, Okay. I'll figure out another way.